libraries/chain/apply_context.cpp

```diff
@@ -301,6 +301,8 @@ void apply_context::schedule_deferred_transaction( const uint128_t& sender_id, a
          });
    }
 
+   EOS_ASSERT( control.is_ram_billing_in_notify_allowed() || (receiver == act.account) || (receiver == payer) || privileged,
+               subjective_block_production_exception, "Cannot charge RAM to other accounts during notify." );
    trx_context.add_ram_usage( payer, (config::billable_size_v<generated_transaction_object> + trx_size) );
```

Comment:
Should this be moved to `trx_context.add_ram_usage` so that all cases are covered in one spot. This seems to miss `eosio_contract.cpp` `apply_eosio_`* call and `transaction_context.schedule_transaction`.

Reply:
The `apply_eosio_*` calls are not a problem since they never run in the context of a notification handler. Similarly with `transaction_context.schedule_transaction` which does not even run in the context of executing an action. The payer in that latter case is the first authorizer of the transaction which needs to be allowed to be billed for the RAM of the delayed transaction.

I didn't put the check in `transaction_context::add_ram_usage` because it doesn't have access to the `apply_context` fields,  such as `receiver` and `act.account`, needed to make the decision of whether or not to throw the `subjective_block_production_exception`.